OCR/frontend/src/componets/ExtractUploadFile.tsx

```diff
@@ -10,15 +10,68 @@ interface IFilesObj {
   files: File[];
 }
 
+interface Template {
+  name: string;
+  description: string;
+  pages: {
+    image: string;
+    fieldNames: string[];
+  }[];
+}
+
 export const ExtractUploadFile: React.FC<ExtractUploadFileProps> = ({
   onUploadComplete,
 }) => {
   const id = useId();
   const { addFile } = useFiles();
   const [template, setTemplate] = useState<string>("");
   const [uploadProgress, setUploadProgress] = useState<number>(0);
+  const [templates, setTemplates] = useState<Template[]>([]);
   const [uploadedFile, setUploadedFile] = useState<File | null>(null);
 
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const loadTemplatesFromLocalStorage = () => {
+    const storedTemplates = localStorage.getItem("templates");
+    if (storedTemplates) {
+      const parsedTemplates = JSON.parse(storedTemplates);
+      setTemplates(parsedTemplates);
+    } else {
+      setTemplates([]);
+    }
+  };
+
+  const loadTemplatesFromLocalStorageTestData = () => {
+    // Sample for testing
+    const sampleTemplates: Template[] = [
+      {
+        name: "Template COVID",
+        description: "This is the first sample template.",
+        pages: [
+          {
+            image: "base64encodedimage1",
+            fieldNames: ["patient_name", "patient_dob"],
+          },
+        ],
+      },
+      {
+        name: "Template Syph",
+        description: "This is the second sample template.",
+        pages: [
+          {
+            image: "base64encodedimage2",
+            fieldNames: ["patient_name", "address"],
+          },
+        ],
+      },
+    ];
+
+    setTemplates(sampleTemplates);
+  };
+
+  useEffect(() => {
```

Comment:
I spoke to @knguyenrise8 he mentioned the templates do not need to be passed in the route from this page.